enhancements/installer/aws-ingress-nlb-security-group.md

```diff
@@ -0,0 +1,498 @@
+- **Maintenance burden**: Maintaining this feature in CCM might become challenging if the upstream cloud-provider-aws project evolves significantly in its load balancer management. Mitigation: Upstream the changes to benefit from community maintenance and align with the long-term strategy for load balancer controllers in OpenShift.
+- **IAM Permissions(?)**: Incorrectly configured IAM permissions for the CCM could lead to failures in creating or managing security groups. Mitigation: Provide clear documentation on the necessary IAM permissions and potentially include checks in the CCM to verify these permissions.
+
+
+### Drawbacks
+
+> WIP/TBReviewed
+
+- The short-term solution requires engineering effort to implement and stabilize the changes in CCM and other components.
+- The extent of changes in CCM may require significant engineering effort from Red Hat to maintain this functionality. To mitigate it, this feature proposes minimum changes without disrupting existing flow.
+- This approach duplicates some functionality that already exists in the AWS Load Balancer Controller.
+
+## Alternatives (Not Implemented)
+
+> TODO/TBD
+
+### **Defaulting to AWS Load Balancer Controller (ALBC) for the default router**:
+
+While ALBC provides more comprehensive support for NLB security groups, this option was deemed out of scope for the initial goal of minimal changes and addressing immediate customer needs for security issues within the existing CCM framework.
+
+Migrating to ALBC would involve more significant architectural changes and potentially impact existing deployments. However, this remains a viable long-term strategy.
+
+Here is an overall effort:
+- There might need to work out how we manage ALBC: ALBO will be used, or CCM will manage ALBC? Neither ALBO nor ALBC is in payload at this time. Moving either one into payload requires migrating from CPaaS to Prow, requiring approval to add a new component to the core payload; When ALBO was created, the messaging was that all new components should be addons, no new components in the core payload.
+- Migrating from CCM to ALBC is going to require either disruption for customer workloads when delete and recreate the LB, or a huge investment in effort from engineering to re-architect the way is managed router deployments, LBs, and DNS to enable having two ELBs in parallel for the same router deployment, bleeding traffic over, and then deleting the old ELB.
+- Red Hat might need to continue to support using CCM indefinitely for customers who are unwilling to do this migration, so this isn't necessarily just a one-release migration; it would most likely end up supporting all these configurations (CLB with CCM, NLB with CCM, NLB with ALBC) as well as the migration process in perpetuity.
+- Currently there were only scratched the surface of special cases, such as custom security groups, custom DNS, or potential regressions going from CCM to ALBC.
```

Comment:
i might change this to say "The previous points have only scratched the surface of special cases, ..."

Reply:
Good suggestions. Fixed, thanks!